src/app/components/cart/cart.component.ts

```diff
@@ -1,57 +1,73 @@
-import { Component, EventEmitter, OnInit, Output } from '@angular/core';
+import { Component, OnDestroy, OnInit } from '@angular/core';
+
 import { CartService } from '../../services/cart.service';
-import { IWatch } from '../../app.models';
+import { ICart, IItem } from '../../app.models';
+import { takeUntil } from 'rxjs/operators';
+import { Subject } from 'rxjs';
 
 @Component({
     selector: 'app-cart',
     templateUrl: './cart.component.html',
     styleUrls: ['./cart.component.scss']
 })
-export class CartComponent implements OnInit {
+export class CartComponent implements OnInit, OnDestroy {
 
-    @Output()
-    public showCartEmit: EventEmitter<boolean> = new EventEmitter<boolean>();
+    public items: Map<number, ICart> = new Map<number, ICart>();
 
-    public cartMap: Map<IWatch, number> = new Map<IWatch, number>();
+    public itemsList: Array<IItem> = [];
 
-    public watchesList: Array<IWatch> = [];
+    private destroy$: Subject<void> = new Subject();
 
-    constructor(private cartService: CartService) {
-    }
+    constructor(
+        private cartService: CartService
+    ) {}
 
     public ngOnInit(): void {
-        this.cartMap = this.cartService.getCartMap();
-        this.watchesList = Array.from(this.cartMap.keys());
+        this.cartService.items$
+            .pipe(takeUntil(this.destroy$))
+            .subscribe((items: Map<number, ICart>) => {
+                this.items = items;
+                this.itemsList = Array.from(this.items.values()).map(item => item.item);
+            })
+        ;
+    }
+
+    public ngOnDestroy(): void {
+        this.destroy$.next();
+        this.destroy$.complete();
     }
 
     public closeCart(): void {
-        this.showCartEmit.emit(false);
+        this.cartService.close();
     }
 
-    public reduceWatchesCount(watch: IWatch): void {
-        this.cartService.deleteWatchFromCart(watch, false);
+    public reduceItemsCount(id: number): void {
+        this.cartService.reduceCountItem(id);
```

Comment:
rename reduceCountItem to reduceItemCount to achieve the same naming both in component and service